network/src/noise/stream.rs

```diff
@@ -112,7 +112,7 @@ where
                         Ok(Some(frame_len)) => {
                             // Empty Frame
                             if frame_len == 0 {
-                                self.read_state = ReadState::Init;
```

Comment:
Yeah, this seemed like it wasn't a good idea to not just end for an "empty frame" anyways.  Can you write a unit test for this?

Reply:
sure thing!

Reply:
done, also I changed 

```
self.read_state = ReadState::Eof(Ok(()));
```

to

```
self.read_state = ReadState::Eof(Err(()));
```

which I think is more sensical